.github/workflows/continuous-integration-workflow-hpx.yml

```
@@ -0,0 +1,76 @@
+name: github-Linux-hpx
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${ {github.event_name }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+
+jobs:
+  hpx:
+    name: hpx
+    runs-on: [ubuntu-latest]
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2.2.0
+        with:
+          path: kokkos
+      - name: setup hpx dependencies
+        run:  sudo apt update && sudo apt install hwloc libboost-all-dev clang
+      - name: checkout hpx
+        uses: actions/checkout@v2.2.0
+        with:
+          repository: STELLAR-GROUP/hpx
+          ref: 1.7.1
+          path: hpx
+      - uses: actions/cache@v2
+        id:   cache-hpx
+        with:
+          path:         ./hpx/install
+          key:          kokkos-hwloc-${{ github.ref }}-${{ github.sha }}
+          restore-keys: kokkos-hwloc-${{ github.ref }}
+      - name: configure hpx
+        if: steps.cache-hpx.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p hpx/{build,install}
+          cd hpx/build
+          cmake \
+            -DCMAKE_INSTALL_PREFIX=$PWD/../install \
```

I would recommend this simply to enable assertions within HPX, though tests will run more slowly:
```suggestion
            -DCMAKE_BUILD_TYPE=Debug \
            -DCMAKE_INSTALL_PREFIX=$PWD/../install \
```